lib/org/openpsa/sales/sales/pdf.php

```diff
@@ -0,0 +1,101 @@
+<?php
+/**
+ * @package org.openpsa.sales
+ * @copyright CONTENT CONTROL http://www.contentcontrol-berlin.de/
+ * @license http://www.gnu.org/licenses/gpl.html GNU General Public License
+ */
+
+/**
+ * PDF Manager
+ *
+ * @package org.openpsa.sales
+ */
+class org_openpsa_sales_sales_pdf
+{
+    /**
+     * @var org_openpsa_sales_salesproject_dba
+     */
+    private $salesproject;
+
+    public function __construct(org_openpsa_sales_salesproject_dba $salesproject)
+    {
+        $this->salesproject = $salesproject;
+    }
+
+    /**
+     * @return midcom_db_attachment|NULL
+     */
+    public function get_attachment($autocreate = false)
+    {
+        $pdf_files = org_openpsa_helpers::get_dm2_attachments($this->salesproject,"pdf_file");
+        if (!empty($pdf_files)) {
+            return reset($pdf_files);
+        }
+        if (!$autocreate) {
+            return null;
+        }
+        return $this->render_and_attach();
+    }
+
+    public function get_button_options()
+    {
+        if (empty($message)) {
```

Comment:
the `$message` parameter is not defined, so this will always evaluate to `true`, meaning that the rest of the function is unreachable